core/math/math_funcs.h

```diff
@@ -219,15 +219,15 @@ class Math {
 
 	static _ALWAYS_INLINE_ int wrapi(int value, int min, int max) {
 		int rng = max - min;
-		return min + ((((value - min) % rng) + rng) % rng);
+		return (rng != 0) ? min + ((((value - min) % rng) + rng) % rng) : min;
 	}
 	static _ALWAYS_INLINE_ double wrapf(double value, double min, double max) {
 		double rng = max - min;
-		return value - (rng * Math::floor((value - min) / rng));
+		return (rng != 0) ? value - (rng * Math::floor((value - min) / rng)) : min;
```

Comment:
Isnt the dividing floating point value to zero just return NaN ? Also, I think, comparing double with zero should be with defined precision

Reply:
Ah, nvm rng is downcasted to integer in that case..

Reply:
I didn't pay attention to it, should the comparison use a double here instead? Otherwise, a range of `0.4` could cause the condition to become false.

Reply:
Ah, true if you downcast it to integer it will not work on numbers below 1.0 and will always return a minimum

Reply:
so for example wrapf(x, 0.2, 0.4) - no matter what x is it will return 0.2

Reply:
I guess comparing it with epsilon is enough for right fix

Reply:
Interestingly, `wrapf(26.18, 0.2, 0.4)` returns 0.38` with the current implementation (not using epsilon comparison).

I changed it to use `is_equal_approx()` for clarity's sake anyway.

Reply:
It may return any value from min to max including 0.38, 0.39 or 0.399999